api/src/main/java/org/openmrs/module/reporting/common/SqlRunner.java

```diff
@@ -144,6 +144,54 @@ public SqlResult executeSql(String sql, Map<String, Object> parameterValues) {
         return result;
 	}
 
+    /**
+     * Executes a Sql Script
+     */
+    public SqlIterator executeSqlToIterator(String sql) {
+
+        SqlIterator iterator = null;
+        log.info("Executing SQL...");
+
+        List<String> sqlStatements = new ArrayList<String>();
+        sqlStatements.addAll(parseSqlIntoStatements(sql));
+
+        Boolean originalAutoCommit = null;
+
+        try {
+            originalAutoCommit = connection.getAutoCommit();
+            connection.setAutoCommit(false);
+
+            for (String sqlStatement : sqlStatements) {
+                Statement statement = null;
+                try {
+                    statement = connection.createStatement();
+                    statement.setFetchSize(10);
+                    log.debug("Executing: " + sqlStatement);
+                    statement.execute(sqlStatement);
+                    ResultSet resultSet = statement.getResultSet();
+
+                    if (resultSet != null) {
+                        ResultSetMetaData rsmd = resultSet.getMetaData();
+                        iterator = new SqlIterator(rsmd, resultSet, connection, statement);
+                    }
+                }
+                catch (Exception e) {
+                    closeStatement(statement);
+                    throw e;
+                }
+            }
+            rollback(); // Always rollback, as this is only intended to support querying
```

Comment:
Do you need to rollback with `executeQuery`?

Reply:
I also don't see the need.
I based this code on the function above `executeSql()` and they seem to keep even if they are only querying.
But I will remove and test it.